test/python/circuit/test_initializer.py

```diff
@@ -314,6 +314,25 @@ def test_equivalence(self):
 
         self.assertEqual(qc1, qc2)
 
+    def test_ndarray(self):
```

Comment:
This test doesn't really do anything. The issue this is fixing is specifically with Aer so won't be caught by basic aer (initialize is unrolled to gates for BasicAer).

A better test would be to just create an instruction for `initialize` and `diagonal` and check the type of the params list. Eg. something like

```python
params = DiagonalGate(np.array([1, 1, 1, 1j]).params
for param in params:
    self.assertFalse(isinstance(param, np.number))
```

Reply:
you are right. Following your suggesting in 9752a6f46bf457af6297edbf05899f65ffb130df

Adding something similar to `DiagonalGate` in 5c848808a